fe/fe-core/src/main/java/org/apache/doris/datasource/hive/source/HiveScanNode.java

```diff
@@ -217,6 +229,46 @@ protected List<Split> getSplits() throws UserException {
         }
     }
 
+    public List<Split> getNextBatch(int maxBatchSize) throws UserException {
+        try {
+            HiveMetaStoreCache cache = Env.getCurrentEnv().getExtMetaCacheMgr()
+                    .getMetaStoreCache((HMSExternalCatalog) hmsTable.getCatalog());
+            String bindBrokerName = hmsTable.getCatalog().bindBrokerName();
+            List<Split> allFiles = Lists.newArrayList();
+            int numPartitions = 0;
+            while (allFiles.size() < maxBatchSize && prunedPartitionsIter.hasNext()) {
+                List<HivePartition> partitions = new ArrayList<>(NUM_PARTITIONS_PER_LOOP);
+                for (int i = 0; i < NUM_PARTITIONS_PER_LOOP && prunedPartitionsIter.hasNext(); ++i) {
+                    partitions.add(prunedPartitionsIter.next());
+                    numPartitions++;
+                }
+                getFileSplitByPartitions(cache, partitions, allFiles, bindBrokerName);
+            }
+            if (allFiles.size() / numPartitions > numSplitsPerPartition) {
+                numSplitsPerPartition = allFiles.size() / numPartitions;
+            }
+            return allFiles;
+        } catch (Throwable t) {
+            LOG.warn("get file split failed for table: {}", hmsTable.getName(), t);
+            throw new UserException(
+                    "get file split failed for table: " + hmsTable.getName() + ", err: " + Util.getRootCauseMessage(t),
+                    t);
+        }
+    }
+
+    public boolean hasNext() {
+        return prunedPartitionsIter.hasNext();
+    }
+
+    public boolean isBatchMode() {
+        int numPartitions = ConnectContext.get().getSessionVariable().getNumPartitionsInBatchMode();
```

Comment:
Is it possible that that there are few partitions(<= 1024), but each partition has a lot of files?

Reply:
When stable, the getSplits() interface will be removed, only the batch interface will be used